compiler/rustc_target/src/spec/mod.rs

```diff
@@ -2452,20 +2452,20 @@ impl Default for TargetOptions {
             static_position_independent_executables: false,
             plt_by_default: true,
             relro_level: RelroLevel::None,
-            pre_link_objects: Default::default(),
-            post_link_objects: Default::default(),
-            pre_link_objects_self_contained: Default::default(),
-            post_link_objects_self_contained: Default::default(),
+            pre_link_objects: MaybeLazy::lazy(Default::default),
```

Comment:
Maybe keep the `Default` impls?
I don't see why this needs to be changed now, if we ever get to constructing targets at compile time, const traits should also be ready at that point.